test/e2e/utils/recordedit-helpers.js

```diff
@@ -613,6 +613,9 @@ exports.testPresentationAndBasicValidation = function(tableParams, isEditMode) {
                                     }).then(function(ct) {
                                         expect(ct).toBe(col.count, colError(col.name, "number of foreign key rows are not as expected."));
 
+                                        return chaisePage.recordsetPage.getTotalCount().getText();
+                                    }).then(function(text) {
+                                        expect(text).toBe("Displaying 5 of " + col.count + " Records", "The total count display in the foreign key popup is incorrect");
```

Comment:
The way I wrote the test cases `col.count` was showing the number of rows that are visible. So it's not actually the total number. This shouldn't be a problem if the count is less than 25 (maximum page size allowed).  You can make a note of it in the jsdoc of this function, or add another `totalCount` variable.
Also change the exception message to use `colError`  which will add column name and index.

```javascript
// if you're adding the note that count should not be more than 25.
var expected = "Displaying " + col.count + " of " + col.count + " Records";

// if you're adding the new variable
var expected = "Displaying " + col.count + " of " + col.totalCount   + " Records";

expect(text).toBe(expected, colError(col.name, "The total count display in the foreign key popup is incorrect");
```
Is the maximum number of foreignkeys 25? if it's not please change it accordingly.